lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -107,6 +108,25 @@ def available_version
     version_field.try(:fetch_path, 'field', 'default_value')
   end
 
+  def self.import_git_url(options)
+    gr = GitRepository.find_or_create_by(:url => options['url'])
+    if options['userid'] && options['password']
+      auth = gr.authentications.detect { |item| item.authtype == 'default' }
```

Comment:
There is a method that does this already... `authentication_default`, I believe.